### PR TITLE
Fix Hive bucket calculation for timestamps with milliseconds

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveBucketing.java [hunk 1]
```diff
@@ -53,82 +49,84 @@
...
-        assertBucketEquals("array<string>", ImmutableList.of("test1", "test2", "test3", "test4"));
-        assertBucketEquals("map<float,date>", null);
-        assertBucketEquals("map<double,timestamp>", ImmutableMap.of());
-        assertBucketEquals("map<string,bigint>", ImmutableMap.of("key", 123L, "key2", 123456789L, "key3", -123456L));
-        assertBucketEquals("array<array<bigint>>", ImmutableList.of(ImmutableList.of(10L, 20L), ImmutableList.of(-10L, -20L), asList((Object) null)));
-        assertBucketEquals("map<array<double>,map<int,timestamp>>", ImmutableMap.of(ImmutableList.of(12.3, 45.7), ImmutableMap.of(123, new Timestamp(1_234_567_890_000L))));
+        assertBucketEquals("boolean", null, 0);
+        assertBucketEquals("boolean", true, 1);
+        assertBucketEquals("boolean", false, 0);
+        assertBucketEquals("tinyint", null, 0);
```
Comment (findepi) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveBucketing.java]: > `Fix Hive bucket calculation for timestamps with milliseconds`

Extract changes of `assertBucketEquals` to a prep commit.

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveBucketing.java [hunk 2]
```diff
@@ -53,82 +49,84 @@
...
+                95411006);
         assertBucketEquals(
                 ImmutableList.of("double", "array<smallint>", "boolean", "map<string,bigint>", "tinyint"),
-                asList(null, ImmutableList.of((short) 5, (short) 8, (short) 13), null, ImmutableMap.of("key", 123L), null));
+                asList(null, ImmutableList.of((short) 5, (short) 8, (short) 13), null, ImmutableMap.of("key", 123L), null),
+                154207826);
     }
 
-    private static void assertBucketEquals(String hiveTypeStrings, Object hiveValues)
+    private static void assertBucketEquals(String hiveTypeStrings, Object hiveValues, int expectedHashCode)
```
Comment (findepi) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveBucketing.java]: hiveTypeStrings -> hiveTypeString
hiveValues -> hiveValue

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveBucketing.java [hunk 3]
```diff
@@ -78,12 +78,12 @@ public static int getHiveBucket(int bucketCount, List<TypeInfo> types, Object[]
         return getBucketNumber(getBucketHashCode(types, values), bucketCount);
     }
 
-    private static int getBucketNumber(int hashCode, int bucketCount)
+    public static int getBucketNumber(int hashCode, int bucketCount)
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/HiveBucketing.java]: VisibleForTesting and package private ?

(here & below)

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveBucketing.java (hunk 1 above)
Comment (electrum) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveBucketing.java]: Thanks, I actually did have this separate, but I somehow lost it while rebasing. `git reflog` saves the day.

**Pull request merged**